data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/cnec/FlowCnecCreator.java

```diff
@@ -183,101 +188,61 @@ private Double getCurrentLimitFromBranch(Branch<?> branch, TwoSides side) {
         cnecAdder.add();
-        if (hasNoPatl) {
-            String cnecName = getCnecName(instantId, null);
-            csaProfileCnecCreationContexts.add(CsaProfileElementaryCreationContext.imported(nativeAssessedElement.mrid(), cnecName, cnecName, "the AssessedElement was pointing to a TATL and used inBaseCase. For the preventive instant, this TATL was also used as a PATL to create the CNEC", true));
-            return;
-        }
-        markCnecAsImportedAndHandleRejectedContingencies(getCnecName(instantId, contingency));
     }
 
-    private void addCurativeFlowCnec(Branch<?> networkElement, Contingency contingency, String instantId, EnumMap<TwoSides, Double> thresholds, boolean useMaxAndMinThresholds, int tatlDuration) {
-        if (thresholds.isEmpty()) {
-            return;
-        }
-        FlowCnecAdder cnecAdder = initFlowCnec();
-        addCnecBaseInformation(cnecAdder, contingency, instantId, tatlDuration);
-        for (TwoSides side : thresholds.keySet()) {
-            double threshold = thresholds.get(side);
-            addFlowCnecThreshold(cnecAdder, side == TwoSides.ONE ? Side.LEFT : Side.RIGHT, threshold, useMaxAndMinThresholds);
-        }
-        cnecAdder.withNetworkElement(networkElement.getId());
-        setNominalVoltage(cnecAdder, networkElement);
-        setCurrentLimitsFromBranch(cnecAdder, networkElement);
-        cnecAdder.add();
-        markCnecAsImportedAndHandleRejectedContingencies(getCnecName(instantId, contingency, tatlDuration));
-    }
-
-    private void addAllFlowCnecsFromBranchAndOperationalLimits(Branch<?> networkElement, Map<Integer, EnumMap<TwoSides, Double>> thresholds, boolean useMaxAndMinThresholds, boolean definedWithConductingEquipment) {
-        EnumMap<TwoSides, Double> patlThresholds = thresholds.get(Integer.MAX_VALUE);
-        boolean hasPatl = thresholds.get(Integer.MAX_VALUE) != null;
-
+    private void addAllFlowCnecsFromBranchAndOperationalLimits(Branch<?> networkElement, Map<Integer, Map<TwoSides, Double>> thresholds, boolean useMaxAndMinThresholds) {
+        // Preventive CNEC
         if (nativeAssessedElement.inBaseCase()) {
-            // If no PATL, we use the lowest TATL instead (as in PowSyBl).
-            if (hasPatl) {
-                addFlowCnec(networkElement, null, crac.getPreventiveInstant().getId(), patlThresholds, useMaxAndMinThresholds, false);
-            } else if (definedWithConductingEquipment) {
-                // No PATL thus the longest acceptable duration is strictly lower than Integer.MAX_VALUE
-                Optional<Integer> longestAcceptableDuration = thresholds.keySet().stream().max(Integer::compareTo);
-                longestAcceptableDuration.ifPresent(integer -> addFlowCnec(networkElement, null, crac.getPreventiveInstant().getId(), thresholds.get(integer), useMaxAndMinThresholds, true));
+            thresholds.getOrDefault(Integer.MAX_VALUE, Map.of()).forEach((twoSides, threshold) -> {
```

Comment:
In PowSyBl, if a line a operational limits, there must necessarily be a PATL. A problem may occur if no operational limit was defined at all for the line. But in that case, as we iterate over the thresholds map, the code will just throw an exception since the importer will try to create a FlowCNEC without threshold.

Reply:
I will add an exception to be added to the CracCreationContext when a line has no operational limit.

Reply:
Ok that sounds good and the log is clear